internal/node/controller/server.go

```diff
@@ -159,6 +160,90 @@ func (s *CtrlServer) GetNodeState(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *CtrlServer) AddUser(w http.ResponseWriter, r *http.Request) {
+	var req atproto.ServerCreateAccount_Input
+	slurp, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err = json.Unmarshal(slurp, &req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	out, err := s.inner.addUser(r.Context(), &req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	bytes, err := json.Marshal(out)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if _, err := w.Write(bytes); err != nil {
+		log.Err(err).Msgf("error sending response in for AddUser request")
+	}
+}
+
+type MigrateRequest struct {
+	TargetVersion string `json:"target_version"`
+}
+
+func (s *CtrlServer) MigrateDB(w http.ResponseWriter, r *http.Request) {
+	var req MigrateRequest
+	slurp, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err = json.Unmarshal(slurp, &req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
maybe time to make this part a helper function ... @sashankg how does glean make it look nice?

Reply:
yeah we have wrappers for each error status that take a internal and user facing error

Reply:
imo leave as is for now until we have a better story for how to log and track errors